### DISCUSSION TITLE
Correct JIT value

### DISCUSSION THREAD

**Discussion by adripo** [nextcloud/all-in-one#4416] [Ideas] [answered]
I was reviewing the correct configuration for OPcache and JIT and came across the following message.

Do we have any update on this checks? Should we disable JIT completely as suggested or stick with the new [`8M`](https://github.com/nextcloud/all-in-one/blob/main/Containers/nextcloud/Dockerfile#L107) value?

https://github.com/nextcloud/all-in-one/issues/2755#issuecomment-1591848523

<blockquote>
              I set up a test instance with the exact same apps enabled and JIT. Already before having all apps enabled, after having JIT enabled, while browsing through the Nextcloud UI, the OPcache usage went up step by step up to max 64 MiB (our installer limits it). Reducing the JIT buffer size to 4 MiB did not help, but disabling it did help.

So with:
- JIT disabled
- Nextcloud 26.0.2 with apps matching https://github.com/nextcloud/all-in-one/issues/2755#issuecomment-1591644114
- OPcache usage 45.23 MiB
- 2350 files cached

With:
- JIT enabled with 2 MiB size
- OPcache usage after browsing through only the apps: 56.37 MiB, 2016 files cached
- OPcache usage after finished browsing through all apps and settings pages, matching the test without JIT: 62.43 MiB, 2350 files cached

So I was able to bring the exact same 2350 files into the cache, but with JIT enabled, even with only 2 MiB size (only 1 MiB is shown as used anyway), the OPcache usage went up significantly.

I suspect the following:
- When I first enabled JIT with PHP 8.0 on our server, I configured a significantly larger JIT size, not sure whether 128 MiB or 64 MiB or different, but I know that I wouldn't have set it to anything significantly larger than the reported usage.
- When I reviewed OPcache stats some time later, only 1 MiB was used, and I hence reduced JIT size to 2 MiB and was wondering that time why JIT usage is only so low. I thought probably it has been enhanced, better garbage collection or such, and was not further thinking about it.
- Now observing this increased OPcache usage, I guess that JIT still has the same memory usage, but allocates the major part of it from the overall OPcache, instead of from the dedicated JIT buffer size. Strange, intransparent, quite problematic as the JIT buffer size then has zero meaning, but this would explain things quite well.

@szaimen 
Reducing the JIT buffer size to 4 MiB does not hurt, but as of above it does not help either. As JIT is a great performance boost, it is probably better to instead increase OPcache memory usage (and interned strings buffer to mute its warning in all cases).

But for testing, probably you can generate a testing container image from a feature branch with JIT disabled, or disable it right within the container image via `opcache.jit=off` and see whether you can replicate my test results. If so, I would report/ask about this at the [PHP bug tracker](https://bugs.php.net/). When on it, I can ask about the constantly increasing interned strings buffer in same cases, how to get more info/debug it etc.

_Originally posted by @MichaIng in https://github.com/nextcloud/all-in-one/issues/2755#issuecomment-1591848523_
            
</blockquote>

**Comment by szaimen** [comment, marked as answer]
Hi, what exactly do you want me to do here?

I already decided what to do here: https://github.com/nextcloud/all-in-one/issues/2755#issuecomment-1592843406

Here is the pr in question: https://github.com/nextcloud/all-in-one/pull/2762

**Comment by adripo** [comment]
Hey @szaimen, thanks for your reply.
I saw the PR and your choices. I was more curious about disabling opcache JIT completely with `opcache.jit=off`.
@MichaIng suggested testing it to see if it's better with or without JIT. I just wanted to know if there was any update on this.

**Reply by szaimen** [reply]
Ah i see. Yeah decided against it and chose another route
